statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/StatefulFunctionsConfig.java

```diff
@@ -0,0 +1,165 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.statefun.flink.core;
+
+import static org.apache.flink.configuration.description.TextElement.code;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.description.Description;
+import org.apache.flink.statefun.flink.core.message.MessageFactoryType;
+import org.apache.flink.statefun.sdk.spi.StatefulFunctionModule;
+import org.apache.flink.util.InstantiationUtil;
+
+/** Configuration that captures all stateful function related settings. */
+@SuppressWarnings("WeakerAccess")
+public class StatefulFunctionsConfig implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+
+  private static final String MODULE_CONFIG_PREFIX = "statefun.module.config.";
+
+  // This configuration option exists for the documentation generator
+  @SuppressWarnings("unused")
+  public static final ConfigOption<String> MODULE_GLOBAL_DEFAULT =
+      ConfigOptions.key(MODULE_CONFIG_PREFIX + "<KEY>")
+          .stringType()
+          .noDefaultValue()
+          .withDescription(
+              Description.builder()
+                  .text(
+                      "Adds the given key/value pair to the Stateful Functions global configuration.")
+                  .text(
+                      "These values will be available via the `globalConfigurations` parameter of StatefulFunctionModule#configure.")
+                  .linebreak()
+                  .text(
+                      "Only the key <KEY> and value are added to the configuration. If the key/value pairs")
+                  .list(
+                      code(MODULE_CONFIG_PREFIX + "key1: value1"),
+                      code(MODULE_CONFIG_PREFIX + "key2: value2"))
+                  .text("are set, then the map")
+                  .list(code("key1: value1"), code("key2: value2"))
+                  .text("will be made available to your module at runtime.")
+                  .build());
+
+  public static final ConfigOption<MessageFactoryType> USER_MESSAGE_SERIALIZER =
+      ConfigOptions.key("statefun.message.serializer")
```

Comment:
No, the prefix is only for specified `globalConfigurations` passed into the module. Before the `StatefulFunctionsModule#configure` was getting the entire flink conf. That is dangerous as it exposes way to many internal details and potentially breaking changes since lots of internal configurations are added between the dispatcher and the TM where we were previously pulling from. 

What I updated this to do was only pass through flink-confs with the prefix `statefun.module.config`. I'm going to update it to be `statefun.module.global-config` instead.

Reply:
Take a look at `StatefulFunctionsConfigTest` and my doc updates